lib/Dialect/Gluon/Transforms/InferCoalescedEncodings.cpp

```diff
@@ -0,0 +1,121 @@
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/Visitors.h"
+#include "triton/Analysis/AxisInfo.h"
+#include "triton/Dialect/Gluon/Transforms/InferLayoutUtils.h"
+#include "triton/Dialect/Gluon/Transforms/Passes.h"
+#include "triton/Dialect/Triton/IR/Utility.h"
+#include "triton/Dialect/TritonGPU/IR/Dialect.h"
+#include "triton/Dialect/TritonGPU/Transforms/CoalesceUtils.h"
+#include "triton/Dialect/TritonGPU/Transforms/Utility.h"
+#include "triton/Tools/StrUtil.h"
+#include "llvm/ADT/PriorityWorklist.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/xxhash.h"
+
+#define DEBUG_TYPE "gluon-infer-coalesced-encodings"
+#define DBGS() (llvm::dbgs() << "[" DEBUG_TYPE "]: ")
+#define LDBG(X) LLVM_DEBUG(DBGS() << X << "\n")
+
+namespace ttg = mlir::triton::gpu;
+
+namespace mlir::triton::gluon {
+
+#define GEN_PASS_DEF_GLUONINFERCOALESCEDENCODINGSPASS
+#include "triton/Dialect/Gluon/Transforms/Passes.h.inc"
+
+namespace {
+
+ttg::CTALayoutAttr getDefaultCTALayout(RankedTensorType refTensorType,
+                                       int numCTAs) {
+  // TODO support numCTAs > 1
+  assert(numCTAs == 1 && "only numCTAs == 1 is supported for now");
+  return ttg::CTALayoutAttr::getDefault(refTensorType.getContext(),
+                                        refTensorType.getShape().size());
+}
+
+bool isCoalescedEncodingTensorType(Type ty) {
+  auto tensorTy = dyn_cast<RankedTensorType>(ty);
+  return tensorTy && isa<gluon::CoalescedEncodingAttr>(tensorTy.getEncoding());
+}
+
+LogicalResult InferCoalescedLayout(ModuleOp &mod,
+                                   llvm::function_ref<bool(Type)> typeCheck) {
```

Comment:
1. Casing is wrong, 2. why parameterize `typeCheck` when it's always `isCoalescedEncodingTensorType` and the code assumes it always is.
```suggestion
LogicalResult inferCoalescedLayout(ModuleOp &mod) {
```